agent/doc/UML/test_runner_design.puml

```diff
@@ -75,20 +75,24 @@ deactivate TaskCompletion
     "maxStepCount": "",
     "deviceTestCount": "",
     "testTimeOutSec": "",
-    "deviceSetUpActions":[{
-            "action":"setProperty",
-            "args": ["log.tag.WelcomeScreen","Verbose"]
+    "deviceActions":[{
+            "timingToAct":"setUp",
+            "actionOrder":"0",
+            "actionType":"setProperty",
+            "actionArgs": ["log.tag.WelcomeScreen","Verbose"]
         },
-        {
-            "action":"setProperty",
-            "args": ["log.tag.WelcomeScreen","Verbose"]
+        {   "timingToAct":"setUp",
+            "actionOrder":"0",
+            "actionType":"setProperty",
```

Comment:
please add device_type, options could include android/ios/windows to represent different type of device managers, and by default it's android.